lib/theme_check/node.rb

```diff
@@ -132,12 +162,81 @@ def position
       )
     end
 
+    def start_token
+      return "" if inside_liquid_tag?
+      output = ""
+      output += "{{" if variable?
+      output += "{%" if tag?
+      output += "-" if whitespace_trimmed_start?
+      output
+    end
+
+    def end_token
+      return "" if inside_liquid_tag?
+      output = ""
+      output += "-" if whitespace_trimmed_end?
+      output += "}}" if variable?
+      output += "%}" if tag?
+      output
+    end
+
     def start_index
       position.start_index
     end
 
     def end_index
       position.end_index
     end
+
+    # Here we're hacking around a glorious bug in Liquid that makes it so the
+    # line_number and markup of a tag is wrong if there's whitespace
+    # between the tag_name and the markup of the tag.
```

Comment:
Do you have a sense if this is fixable in Liquid, or is just the way it works?

Reply:
I think it would be fixable but the way tags are created you don't really know. We'd need to somehow pass it in the parse_context since we lose the information on tag creation (the tag_name and markup come in as two separate variables)

https://github.com/Shopify/liquid/blob/eab13a07d9861a38d993d2749ae25f06ff76426b/lib/liquid/tag.rb#L25-L37